compiler/quilt/tools/compat.py

```diff
@@ -0,0 +1,36 @@
+"""
+Python 2 / 3 compatibility
+
+In cases where the `six` lib covers an item, use `six`.
+
+For others, import the module here under the python3 name (if there are naming differences).
+
+Unfortunately, this *doesn't* allow for `from .tools.compat.foo import bar`, so if needed,
+import frequently-used objects here for convenience.
+"""
+# unused imports -- pylint: disable=W0611
+
+import sys as _sys
+
+
+# Reflecting requirements in setup.py
+# Python < 3.4
+if _sys.version_info < (3, 4):
+    from backports import tempfile
+    from funcsigs import signature  # inspect.argspec is deprecated
+else:
+    import tempfile
+    from inspect import signature   # inspect.argspec is deprecated
+
+# Python < 3.5
+if _sys.version_info < (3, 5):
+    import pathlib2 as pathlib
+else:
+    import pathlib
+
+
+# Example convenience references to allow `from .tools.compat import some_obj`
+
+# patch = mock.patch
+# Path = pathlib.Path
+# TemporaryDirectory = tempfile.TemporaryDirectory
```

Comment:
This doesn't seem to be used anywhere. Is it part of TensorFlow?

Reply:
It was, but has been dropped.  Now that the TF Prep PR has been merged, I can merge master into this and clean up, then squash and merge into master.